storage/innobase/lock/lock0lock.cc

```diff
@@ -5553,7 +5553,7 @@ dberr_t lock_sys_t::cancel(trx_t *trx, lock_t *lock, bool check_victim)
 /** Cancel a waiting lock request (if any) when killing a transaction */
 void lock_sys_t::cancel(trx_t *trx)
 {
-  mysql_mutex_lock(&lock_sys.wait_mutex);
+  lock_sys.wait_mutex_lock();
```

Comment:
This function is only executed by `innobase_kill_query` (a.k.a. handling `KILL QUERY` or `KILL CONNECTION`). I do not think that it is time-critical enough to mandate spinning. While we are not holding any InnoDB mutexes here, we probably are (or should be) holding some `THD` mutex; see [MDEV-26014](https://jira.mariadb.org/browse/MDEV-26014).